packages/dx-scheduler-core/src/plugins/resources/computeds.test.ts

```diff
@@ -0,0 +1,209 @@
+import { getAppointmentResources, validateResources, convertResourcesToPlain } from './computeds';
+
+describe('Resources computeds', () => {
+  describe('#validateResources', () => {
+    const baseResources = [
+      { fieldName: 'ownerId', items: [{ id: 0 }] },
+    ];
+    const palette = ['red', 'brown', 'green', 'pink'];
+
+    it('should assign properties according to resource title', () => {
+      const resources = [
+        { fieldName: 'ownerId', title: 'Owner', items: [{ id: 0 }] },
+      ];
+      expect(validateResources(resources, undefined, palette))
+        .toEqual([
+          {
+            fieldName: 'ownerId',
+            allowMultiple: false,
+            isMain: true,
+            title: 'Owner',
+            items: [{
+              id: 0,
+              color: 'red',
+              allowMultiple: false,
+              isMain: true,
+              text: 'Owner',
+              title: 'Owner',
+              fieldName: 'ownerId',
+            }],
+          },
+        ]);
+    });
+    it('should assign properties according to resource text', () => {
+      const resources = [
+        { fieldName: 'ownerId', title: 'Owner', items: [{ id: 0, text: 'Maxim' }] },
+      ];
+      expect(validateResources(resources, undefined, palette))
+        .toEqual([
+          {
+            fieldName: 'ownerId',
+            allowMultiple: false,
+            isMain: true,
+            title: 'Owner',
+            items: [{
+              id: 0,
+              color: 'red',
+              allowMultiple: false,
+              isMain: true,
+              text: 'Maxim',
+              title: 'Owner',
+              fieldName: 'ownerId',
+            }],
+          },
+        ]);
+    });
+    it('should assign the Main flag to single resource', () => {
+      expect(validateResources(baseResources, undefined, palette))
+        .toEqual([
+          {
+            fieldName: 'ownerId',
+            allowMultiple: false,
+            isMain: true,
+            title: 'ownerId',
+            items: [{
+              id: 0,
+              color: 'red',
+              allowMultiple: false,
+              isMain: true,
+              text: 'ownerId',
+              title: 'ownerId',
+              fieldName: 'ownerId',
+            }],
+          },
+        ]);
+    });
+    it('should assign the Main flag to two resource', () => {
```

Comment:
'should not assign the Main flag to two resources'?
Only the first, `ownerId`, resource has the `isMain` prop equal to `true`